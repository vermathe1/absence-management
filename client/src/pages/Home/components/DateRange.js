import React from "react";
import { DateSelector } from "../../../commonComponents";

export const DateRange = ({ selected, setDate }) => {
  return <DateSelector selected={selected} setDate={(date) => setDate(date)} />;
};
