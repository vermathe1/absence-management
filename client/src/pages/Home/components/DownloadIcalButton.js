import React from "react";
import { Button } from "../../../commonComponents";

export const DownlaodIcalButton = ({ onClick, text }) => {
  return <Button onClick={onClick} text={text} />;
};
