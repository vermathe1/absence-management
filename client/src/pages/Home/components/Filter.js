import React from "react";
import { Button } from "../../../commonComponents";
import styles from "../../../styles/home.module.css";

export const Filter = ({ headings, activeTab, onTabClick }) => {
  return (
    <Tab
      headings={headings}
      active={activeTab}
      onTabClick={(tab) => onTabClick(tab)}
    />
  );
};

const Tab = ({ headings, active, onTabClick }) => {
  const handleClick = (heading) => {
    onTabClick(heading);
  };
  return (
    <ul className={styles.tab}>
      {headings.map((heading, index) => {
        return (
          <li
            className={
              active === heading
                ? `${styles.childTab} ${styles.activeTab}`
                : styles.childTab
            }
            key={index}
            onClick={(e) => handleClick(heading)}
          >
            {" "}
            <a className={styles.nodecoration} href="#">
              {heading}
            </a>{" "}
          </li>
        );
      })}
    </ul>
  );
};
