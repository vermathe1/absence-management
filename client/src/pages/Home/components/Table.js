import React from "react";
import ReactTable from "react-table";
import "react-table/react-table.css";

export const Table = ({
  columns,
  data,
  defaultPageSize,
  noDataText,
  filterable,
  pivotBy,
}) => {
  return (
    <ReactTable
      columns={columns}
      data={data}
      defaultPageSize={10}
      noDataText={"Theres no data in the given date!!"}
      filterable
      pivotBy={["userId"]}
    ></ReactTable>
  );
};
