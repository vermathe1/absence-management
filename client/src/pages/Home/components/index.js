export * from "./DateRange";
export * from "./DownloadIcalButton";
export * from "./Filter";
export * from "./Table";
