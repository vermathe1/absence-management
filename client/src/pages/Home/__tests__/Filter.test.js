import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Filter, Tab } from "../components/Filter";

describe("<Filter />", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <Filter
          headings={["a", "b", "c"]}
          activeTab="a"
          onTabClick={jest.fn()}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render Tab with props from Filter", () => {
    const mockFunction = () => null;
    const wrapper = shallow(
      <Filter
        headings={["a", "b", "c"]}
        activeTab="a"
        onTabClick={mockFunction}
      />
    );
    expect(wrapper.find(Tab).props()).toEqual({
      headings: ["a", "b", "c"],
      active: "a",
      onTabClick: mockFunction,
    });
  });
});
describe("<Tab />", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <Tab headings={["a", "b", "c"]} active="a" onTabClick={jest.fn()} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render list of tabs from headings", () => {
    const headings = ["a", "b", "c"];
    const wrapper = shallow(
      <Tab headings={headings} active="a" onTabClick={jest.fn()} />
    );

    expect(wrapper.find("li")).toHaveLength(headings.length);
    headings.forEach((heading, index) => {
      expect(wrapper.find("a").get(index).props.children).toBe(heading);
    });
  });

  it("should call onTabClick when clicked on a Tab", () => {
    const onTabClickMock = jest.fn();
    const headings = ["a", "b", "c"];
    const wrapper = shallow(
      <Tab headings={headings} active="a" onTabClick={onTabClickMock} />
    );
    wrapper.find("li").at(0).simulate("click");
    expect(onTabClickMock).toHaveBeenCalledTimes(1);
    expect(onTabClickMock).toHaveBeenCalledWith(headings[0]);
  });
});
