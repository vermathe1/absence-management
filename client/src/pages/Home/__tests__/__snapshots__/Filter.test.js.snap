// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> should render correctly 1`] = `
<ul
  className="tab"
>
  <li
    className="childTab activeTab"
    onClick={[Function]}
  >
     
    <a
      className="nodecoration"
      href="#"
    >
      a
    </a>
     
  </li>
  <li
    className="childTab"
    onClick={[Function]}
  >
     
    <a
      className="nodecoration"
      href="#"
    >
      b
    </a>
     
  </li>
  <li
    className="childTab"
    onClick={[Function]}
  >
     
    <a
      className="nodecoration"
      href="#"
    >
      c
    </a>
     
  </li>
</ul>
`;

exports[`<Tab /> should render correctly 1`] = `
<ul
  className="tab"
>
  <li
    className="childTab activeTab"
    onClick={[Function]}
  >
     
    <a
      className="nodecoration"
      href="#"
    >
      a
    </a>
     
  </li>
  <li
    className="childTab"
    onClick={[Function]}
  >
     
    <a
      className="nodecoration"
      href="#"
    >
      b
    </a>
     
  </li>
  <li
    className="childTab"
    onClick={[Function]}
  >
     
    <a
      className="nodecoration"
      href="#"
    >
      c
    </a>
     
  </li>
</ul>
`;
